keel-core/src/main/kotlin/com/netflix/spinnaker/keel/serialization/ObjectMappers.kt

```diff
@@ -19,20 +19,15 @@ import de.huxhorn.sulky.ulid.ULID
 import java.text.SimpleDateFormat
 import java.util.TimeZone
 
-private val objectMapperInstance: ObjectMapper by lazy { ObjectMapper().configureMe() }
-private val yamlMapperInstance: YAMLMapper by lazy { YAMLMapper().configureMe() }
-
 /**
  * Factory method for [ObjectMapper]s configured how we like 'em.
  */
-fun configuredObjectMapper(): ObjectMapper =
-  objectMapperInstance
+fun configuredObjectMapper(): ObjectMapper = ObjectMapper().configureMe()
 
 /**
  * Factory method for [YAMLMapper]s configured how we like 'em.
  */
-fun configuredYamlMapper(): YAMLMapper =
-  yamlMapperInstance.disable(USE_NATIVE_TYPE_ID)
+fun configuredYamlMapper(): YAMLMapper = YAMLMapper().configureMe().disable(USE_NATIVE_TYPE_ID)
```

Comment:
This was the ultimate problem. Because `ExceptionHandlerTests` messes with the registration of subtypes on these mappers, and they're package-level variables rather than just singleton Spring beans, the changes that test makes leak into any subsequent tests